File: backend/src/main/kotlin/com/example/backend/config/AppSettingInitializer.kt
package com.example.backend.config

import com.example.backend.models.AppSetting
import com.example.backend.repositories.AppSettingRepository
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Component
class AppSettingInitializer(
  private val appSettingRepository: AppSettingRepository
) : ApplicationRunner {

  override fun run(args: ApplicationArguments?) {
    val defaultSettings = listOf(
      AppSetting(keyName = "app_name", value = "ALSA", type = "string"),
      AppSetting(keyName = "vente_mode", value = "differe", type = "string"),
      AppSetting(keyName = "show_menu_stats", value = "true", type = "boolean")
    )

    defaultSettings.forEach { setting ->
      val existing = appSettingRepository.findByKeyName(setting.keyName!!)
      if (existing == null){
        appSettingRepository.save(setting)
        println("Parametre initialiser : ${setting.keyName} = ${setting.value}")
      }
      else {
        println("Parametre deja present : ${existing.keyName}")
      }
    }
  }

}

File: backend_py/pharmaPython/app/core/app_setting_initializer.py
from app.models.app_setting import AppSetting
from app.repositories import app_setting_repository


def initialize_app_settings(db):
  default_settings = [
    {"key_name": "app_name", "value": "ALSA", "type": "string"},
    {"key_name": "vente_mode", "value": "differe", "type": "string"},
    {"key_name": "show_menu_stats", "value": "true", "type": "boolean"}
  ]

  for s in default_settings:
    existing = db.query(AppSetting).filter(AppSetting.key_name == s["key_name"]).first()
    if existing is None:
      new_setting = AppSetting(**s)
      db.add(new_setting)
      print(f"✅ Paramètre initialisé : {s['key_name']} = {s['value']}")
    else:
      print(f"⚙️ Paramètre déjà présent : {existing.key_name}")
    db.commit()
